xlml/utils/metric.py

```diff
@@ -416,31 +418,53 @@ def get_gce_job_status(benchmark_id: str) -> bigquery.JobStatus:
-  if setup_state == TaskState.SKIPPED.value:
-    logging.info("The setup state is skipped, and the job status is missed.")
-    return bigquery.JobStatus.MISSED
-
-  # check setup status to see if setup step is successful
-  if setup_state == TaskState.FAILED.value:
-    logging.info("The setup state is failed, and the job status is failed.")
+  # GCE SSH method
+  if task_test_config.use_startup_script == False:
+    # check setup status to see if provision step is successful
+    setup_task = current_dag.get_task(task_id=f"{benchmark_id}.provision.setup")
+    setup_ti = TaskInstance(setup_task, execution_date)
+    setup_state = setup_ti.current_state()
+    if setup_state == TaskState.SKIPPED.value:
+      logging.info("The setup state is skipped, and the job status is missed.")
+      return bigquery.JobStatus.MISSED
+
+    # check setup status to see if setup step is successful
+    if setup_state == TaskState.FAILED.value:
+      logging.info("The setup state is failed, and the job status is failed.")
+      return bigquery.JobStatus.FAILED
+
+    # check run_model status to see if run_model step is successful
+    run_model_task = current_dag.get_task(task_id=f"{benchmark_id}.run_model")
+    run_model_ti = TaskInstance(run_model_task, execution_date)
+    run_model_state = run_model_ti.current_state()
+
+    if run_model_state == TaskState.SUCCESS.value:
+      logging.info("The run_model state is success, and the job status is success.")
+      return bigquery.JobStatus.SUCCESS
+
+    logging.info("The run_model state is failed, and the job status is failed.")
     return bigquery.JobStatus.FAILED
-
-  # check run_model status to see if run_model step is successful
-  run_model_task = current_dag.get_task(task_id=f"{benchmark_id}.run_model")
-  run_model_ti = TaskInstance(run_model_task, execution_date)
-  run_model_state = run_model_ti.current_state()
-
-  if run_model_state == TaskState.SUCCESS.value:
-    logging.info("The run_model state is success, and the job status is success.")
-    return bigquery.JobStatus.SUCCESS
-
-  logging.info("The run_model state is failed, and the job status is failed.")
-  return bigquery.JobStatus.FAILED
+  # GCE startup script method
+  else:
+    # check startup_script status to see if startup_script step is successful
+    startup_script_task = current_dag.get_task(
+        task_id=f"{benchmark_id}.provision_with_startup_script.create_queued_resource.check_if_startup_script_end"
+    )
+    startup_script_ti = TaskInstance(startup_script_task, execution_date)
+    startup_script_state = startup_script_ti.current_state()
+    if startup_script_state == TaskState.SKIPPED.value:
+      logging.info("The startup_script state is skipped, and the job status is missed.")
+      return bigquery.JobStatus.MISSED
+    elif startup_script_state == TaskState.FAILED.value:
+      logging.info("The startup_script state is failed, and the job status is failed.")
+      return bigquery.JobStatus.FAILED
```

Comment:
Have you tested this case? Is the metrics result expected?

Reply:
I have my newest update to check if `provision_with_startup_script.create_queued_resource.check_if_startup_script_end` succeed:
https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/83/commits/b214d4906041b3c0c15572609ef40bbb76e1bb45#diff-74b29a75aab611eb1dff01dac1600bc202aaf83cc6376e182e00676aa6ff06ccR478-R485


From my test case, `check_if_startup_script_end` failed but `process_metrics` passed:
https://screenshot.googleplex.com/7XevQ4UqMvN7W7P

I am a bit confused by the behavior: 
`process_metrics` show it failed, but the task was marked succeed, is that expected? 
https://screenshot.googleplex.com/5vvjicG2kpwgCmC

Reply:
I think this makes sense. Post_process did successfully process those metrics, and send `failed` status into BigQuery table. If you go to Airflow UI, you could see the overall test status is failed as one of task failed: https://screenshot.googleplex.com/73UPv4ogbxDmSKJ.

Reply:
If you check the metric in post_process, the `job_status=1` which means this test failed.

```
[2024-01-26, 00:58:15 UTC] {logging_mixin.py:150} INFO - Test run rows: [TestRun(job_history=JobHistoryRow(uuid='268db43e4b9262c23ae67bbed898890650d7c46eb6414a89778f328c09094395', timestamp=datetime.datetime(2024, 1, 26, 0, 58, 15, 702197), owner='Tony C.', job_name='maxtext-nightly-startup-script-v4-8', job_status=1), metric_history=[], metadata_history=[MetadataHistoryRow(job_uuid='268db43e4b9262c23ae67bbed898890650d7c46eb6414a89778f328c09094395', metadata_key='run_id', metadata_value='manual__2024-01-26T00:43:26.611408+00:00'), MetadataHistoryRow(job_uuid='268db43e4b9262c23ae67bbed898890650d7c46eb6414a89778f328c09094395', metadata_key='prev_start_date_success', metadata_value='2024-01-25T23:03:33.670866+00:00'), MetadataHistoryRow(job_uuid='268db43e4b9262c23ae67bbed898890650d7c46eb6414a89778f328c09094395', metadata_key='airflow_dag_run_link', metadata_value='https://d694f58754c647b2970e3d09b76613a4-dot-us-central1.composer.googleusercontent.com/dags/maxtext_nightly_startup_script_example_dag/grid?dag_run_id=manual__2024-01-26T00%3A43%3A26.611408%2B00%3A00&task_id=maxtext-nightly-startup-script-v4-8.post_process.process_metrics')])]
```

Reply:
Nice! Good to know!